src/components/QRShare/QRShareWithDownload/index.native.js

```diff
@@ -1,37 +1,46 @@
-import React, {Component} from 'react';
+import React, {forwardRef, useImperativeHandle, useRef} from 'react';
 import ViewShot from 'react-native-view-shot';
-import {withNetwork} from '@components/OnyxProvider';
 import getQrCodeFileName from '@components/QRShare/getQrCodeDownloadFileName';
 import {qrShareDefaultProps, qrSharePropTypes} from '@components/QRShare/propTypes';
+import useNetwork from '@hooks/useNetwork';
 import fileDownload from '@libs/fileDownload';
 import QRShare from '..';
 
-class QRShareWithDownload extends Component {
-    qrCodeScreenshotRef = React.createRef();
+function QRShareWithDownload({innerRef, ...props}) {
```

Comment:
Same can be done here

Reply:
Answered [here](https://github.com/Expensify/App/pull/32690#discussion_r1420201260)!